src/CommonProviderImplementation/DebugProvidedTypes-modified.fs

```diff
@@ -0,0 +1,556 @@
+                | t when t.IsPointer -> (t.GetElementType() |> toString useFullName) + "*"
+                | :? ProvidedTypeDefinition as t ->
+                    add t
+                    t.Name.Split(',').[0]
+                | t when t.IsGenericType ->
+                    let gtd = t.GetGenericTypeDefinition()
+                    let fullName = getFullName gtd
+                    let args =
+                        if useFullName then
+                            t.GetGenericArguments() 
+                            |> Seq.map (if hasUnitOfMeasure then (fun t -> t.Name) else toString useFullName)
+                        else
+                            t.GetGenericArguments() 
+                            |> Seq.map (fun _ -> "_")
+                    if fullName.StartsWith "System.Tuple`" then
+                        String.concat " * " args
+                    elif fullName.StartsWith "Microsoft.FSharp.Core.FSharpFunc`" then
+                        "(" + (String.concat " -> " args) + ")"
+                    else 
+                        let args = String.concat "," args
+                        let name, reverse = 
+                            match t with
+                            | _ when hasUnitOfMeasure -> toString useFullName t.UnderlyingSystemType, false
+                            | _ when fullName = typedefof<int seq>.FullName -> "seq", true
+                            | _ when fullName = typedefof<int list>.FullName -> "list", true
+                            | _ when fullName = typedefof<int option>.FullName -> "option", true
+                            | _ when fullName = typedefof<int ref>.FullName -> "ref", true
+                            | _ when fullName = typedefof<Async<_>>.FullName -> "async", true
+                            | _ when ns.Contains gtd.Namespace -> gtd.Name, false
+                            | _ -> (if useFullName then fullName else gtd.Name), false
+                        let name = name.Split('`').[0]
+                        if reverse then
+                            args + " " + name 
+                        else
+                            name + "<" + args + ">"
+                | _  ->
+                let fullName = getFullName t
+                match t with 
+                | _ when fullName = "System.Boolean" -> "bool"
+                | _ when fullName = "System.Object" -> "obj"
+                | _ when fullName = "System.Int32" -> "int"
+                | _ when fullName = "System.Int64" -> "int64"
+                | _ when fullName = "System.Double" -> "float"
+                | _ when fullName = "System.Single"-> "float32"
+                | _ when fullName = "System.Decimal" -> "decimal"
+                | _ when fullName = "System.String" -> "string"
+                | _ when fullName = "System.Void" -> "()"
+                | _ when fullName = typeof<unit>.FullName -> "()"
```

Comment:
One is unit, one is void (nb. essentially the same as the printing code that was already there).